modules/face/include/opencv2/face.hpp

```diff
@@ -257,6 +266,15 @@ class CV_EXPORTS_W FaceRecognizer : public Algorithm
     /** @overload */
     virtual int predict(InputArray src) const = 0;
 
+	/** @brief - if implemented - send all result of prediction to collector that can be used for somehow custom result handling
+	@param src Sample image to get a prediction from.
+	@param collector User-defined collector function that is acceptor for all prediction-checking results
+
+	To implement this method u just have to do same internal cycle as in predict(InputArray src, CV_OUT int &label, CV_OUT double &confidence) but
+	not try to get "best@ result, just resend it to caller side with given collector
+	*/
+	CV_WRAP virtual void predict_collect(InputArray src, PredictCollector collector) const = 0;
```

Comment:
since the automated scripting generator (for python java,etc)  can't handle callbacks (save c++11 things), this might have to go without the CV_WRAP tag